### DISCUSSION TITLE
Opening telescope choice in a new window

### DISCUSSION THREAD

**Discussion by vdegenne** [LazyVim/LazyVim#4682] [Q&A]
I'm used to open new buffers in new windows from neo-tree selecting a file and pressing `s` will open the file in a buffer vertically next to the last focused buffer (which is very handy).

Does telescope have similar feature in LazyVim default behaviors, or is it something I need to implement myself?

**Comment by dpetka2001** [comment]
It's `<c-v>`. Take a look at all default [telescope mappings](https://github.com/nvim-telescope/telescope.nvim?tab=readme-ov-file#default-mappings). `<c-t>` for new tab won't work because it's used by Trouble in LazyVim, so you would have to implement that yourself.

**Reply by vdegenne** [reply]
Thanks! With your lead and last time knowledge sharing I could easily find a way to bind the key:

```lua
local actions = require("telescope.actions")

local picker_config = {
  live_grep = {
    mappings = {
      n = {
        ["s"] = actions.select_vertical,
      }
    }
  },
  find_files = {
    mappings = {
      n = {
        ["s"] = actions.select_vertical,
      }
    }
  }
}
```

**Reply by dpetka2001** [reply]
Oh, I thought you just wanted to know the default keymap. You didn't mention about changing it. That's why I didn't respond with a code snippet. Glad you got it working :smile:

**Reply by dpetka2001** [reply]
Also, I believe you can instead use something like 
```lua
defaults = {
  mappings = {
    -- your mapping here
  },
}
```
So, that you don't have to do that for every picker. I hope I'm not mistaken, because I don't use Telescope any more :stuck_out_tongue:

**Reply by vdegenne** [reply]
@dpetka2001 I tried that before but for some reason it doesn't work for the find files picker, I have that in my config:
![image](https://github.com/user-attachments/assets/b0b3f70b-941f-4e41-8a89-2393d32a1d9c)

And when I press `s` in telescope, this happens:

![image](https://github.com/user-attachments/assets/9ad4a52c-4553-4f94-899e-ee0a632dd8d9)

So I guess it's being rewritten by another plugin, I'm not sure which one, is it a flash feature?

**Reply by dpetka2001** [reply]
Yes, this is `flash.nvim`. Like I said I don't use Telescope any more, so was not sure if it was going to work. 

However, you seem to be using `vim.tbl_deep_extend` the wrong way. I believe it should be 
```lua
opts.defaults = vim.tbl_deep_extend("force", opts.defaults, {
  -- your custom settings here
})
```
If that doesn't work, then just keep your initial attempt.

The following seems to work for me
```lua
    opts = function(_, opts)
      local actions = require("telescope.actions")
      opts.defaults = vim.tbl_deep_extend("force", opts.defaults, {
        mappings = {
          n = {
            ["s"] = actions.select_vertical,
          },
        },
      })
    end,
```

**Reply by vdegenne** [reply]
Oh man it's like `Object.assign` in JS, I don't know why I failed that..! Anyway thanks :smile:
  
(I switched back to Telescope because I like Aerial, also for some reason fzf is slower on my oldest computers..)
